### PR TITLE
fix(Group filtering): return user group when no filtered group are available

### LINKED ISSUE #171: Bypass filtering when assigning groups not Working
The existing groups in GLPI are not appearing for assignment, only those to which the technician is linked. Checking "ignore" should appear.
I have been using this resource for a long time e I know this resource.
Plugin Version 2.9.3

### PR DESCRIPTION
**Pull request by stonebuzz**
With ```Enable filtering on the groups assignment``` option  enabled

When no group is assigned to the ticket, the list of actors contains no group (we should have all the user's groups logged in).

as soon as a group is assigned, group filtering works correctly

Fix #171

### CODE REVIEW

**Review by ccailly — APPROVED** [review]

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]
Do we always want to show the users groups when the results are empty ?
IMO If the ticket has a group and this group doesn't allow any escalation then it is correct to show empty results.

However, if this ticket doesn't have a group yet (first assignment), then it would be OK to list all the user groups as possible choices.

What do you think ?

**Comment by stonebuzz** [comment]
> Do we always want to show the users groups when the results are empty ? IMO If the ticket has a group and this group doesn't allow any escalation then it is correct to show empty results.
> 
> However, if this ticket doesn't have a group yet (first assignment), then it would be OK to list all the user groups as possible choices.
> 
> What do you think ?

Agree with that; see my last commit

Waiting for community approval

**Review by Rom1-B — COMMENTED** [review]
Seems OK, but I agree it needs community feedback

File: inc/group_group.class.php
Comment (AdrienClairembault) [inline inc/group_group.class.php:189]: I don't think checking if the ticket is new is the correct way to handle this.
A ticket may be created without groups, or its groups may be deleted after a while.

In both this cases, it seems valid to allow the user to assign any of his groups to the ticket.
I think we should probably check that the tickets groups count is 0.

**Review by AdrienClairembault — CHANGES_REQUESTED** [review]

File: inc/group_group.class.php
Comment (stonebuzz) [reply inc/group_group.class.php:189]: the variable ```$groups``` already refers to the ticket's groups, but only if the ticket exists (```$id > 0```) see ```L144```

**Comment by stonebuzz** [comment]
Close in favor of https://github.com/pluginsGLPI/escalade/pull/176

**Pull request closed without merging**
